i6_models/assemblies/conformer/conformer_v1.py

```diff
@@ -0,0 +1,118 @@
+from __future__ import annotations
+
+__all__ = ["ConformerBlockV1Config", "ConformerEncoderV1Config", "ConformerBlockV1", "ConformerEncoderV1"]
+
+import torch
+from torch import nn
+from dataclasses import dataclass
+from typing import Tuple
+
+from i6_models.config import ModelConfiguration, ModuleFactoryV1
+from i6_models.parts.conformer import (
+    ConformerConvolutionV1,
+    ConformerConvolutionV1Config,
+    ConformerMHSAV1,
+    ConformerMHSAV1Config,
+    ConformerPositionwiseFeedForwardV1,
+    ConformerPositionwiseFeedForwardV1Config,
+)
+
+
+@dataclass
+class ConformerBlockV1Config(ModelConfiguration):
+    """
+    Attributes:
+        ff_cfg: Configuration for ConformerPositionwiseFeedForwardV1
+        mhsa_cfg: Configuration for ConformerMHSAV1
+        conv_cfg: Configuration for ConformerConvolutionV1
+    """
+
+    # nested configurations
+    ff_cfg: ConformerPositionwiseFeedForwardV1Config
+    mhsa_cfg: ConformerMHSAV1Config
+    conv_cfg: ConformerConvolutionV1Config
+
+
+class ConformerBlockV1(nn.Module):
+    """
+    Conformer block module
+    """
+
+    def __init__(self, cfg: ConformerBlockV1Config):
+        """
+        :param cfg: conformer block configuration with subunits for the different conformer parts
+        """
+        super().__init__()
+        self.ff1 = ConformerPositionwiseFeedForwardV1(cfg=cfg.ff_cfg)
+        self.mhsa = ConformerMHSAV1(cfg=cfg.mhsa_cfg)
+        self.conv = ConformerConvolutionV1(model_cfg=cfg.conv_cfg)
+        self.ff2 = ConformerPositionwiseFeedForwardV1(cfg=cfg.ff_cfg)
+        self.final_layer_norm = torch.nn.LayerNorm(cfg.ff_cfg.input_dim)
+
+    def forward(self, tensor: torch.Tensor, sequence_mask: torch.Tensor) -> torch.Tensor:
+        """
+        :param tensor: input tensor of shape [B, T, F]
+        :param sequence_mask: mask tensor where 0 defines positions within the sequence and 1 outside, shape: [B, T]
+        :return: torch.Tensor of shape [B, T, F]
+        """
+        residual = tensor  #  [B, T, F]
+        x = self.ff1(residual)  #  [B, T, F]
+        residual = 0.5 * x + residual  #  [B, T, F]
+        x = self.mhsa(residual, sequence_mask)  #  [B, T, F]
+        residual = x + residual  # [B, T, F]
+        x = self.conv(residual)  #  [B, T, F]
+        residual = x + residual  # [B, T, F]
+        x = self.ff2(residual)  #  [B, T, F]
+        x = 0.5 * x + residual  #  [B, T, F]
```

Comment:
I find the mix of `tensor`, `x` and `residual` somewhat confusing here. Esp you also sometimes use `residual` as input, sometimes `x`.

Why not directly name the input arg `x`, then you don't have `tensor` here at all. You might also want to require it to be a positional arg.

And you can avoid the `residual` completely maybe. E.g. like:

```suggestion
        x = 0.5 * self.ff1(x) + x  #  [B, T, F]
        x = self.mhsa(x, sequence_mask) + x  #  [B, T, F]
        x = self.conv(x) + x  #  [B, T, F]
        x = 0.5 * self.ff2(x) + x  #  [B, T, F]
```
I personally find this more readable. And it's shorter.